src/EditorFeatures/TestUtilities/ChangeSignature/TestChangeSignatureOptionsService.cs

```diff
@@ -4,31 +4,46 @@
 using System.Linq;
 using Microsoft.CodeAnalysis.ChangeSignature;
 using Microsoft.CodeAnalysis.Host.Mef;
-using Microsoft.CodeAnalysis.Notification;
+using Microsoft.CodeAnalysis.Test.Utilities.ChangeSignature;
 
 namespace Microsoft.CodeAnalysis.Editor.UnitTests.ChangeSignature
 {
     [ExportWorkspaceService(typeof(IChangeSignatureOptionsService), ServiceLayer.Default), Shared]
     internal class TestChangeSignatureOptionsService : IChangeSignatureOptionsService
     {
         public bool IsCancelled = true;
-        public int[] UpdatedSignature = null;
+        public AddedParameterOrExistingIndex[] UpdatedSignature = null;
 
         [ImportingConstructor]
         public TestChangeSignatureOptionsService()
         {
         }
 
-        public ChangeSignatureOptionsResult GetChangeSignatureOptions(ISymbol symbol, ParameterConfiguration parameters, INotificationService notificationService)
+        AddedParameterResult IChangeSignatureOptionsService.GetAddedParameter(Document document, int insertPosition)
+        {
+            throw new System.NotImplementedException();
```

Comment:
https://github.com/dotnet/roslyn/issues/41118

Reply:
GetAddedParameter is a primitive operation just adding values to the viewmodel. We still should cover it in unit tests . This could be done together with #41118

Reply:
@jasonmalinowski @ivanbasov The goal was to test the refactoring code path end-to-end, which involves grabbing _some_ option service... How else can we test that end-to-end (other than integration tests where the real option service is available)?

Reply:
I guess I would have imagined that either:

1. We don't trust our integration tests, or don't want to force users to run them, and then as a result we'd have the end-to-end unit test. In this case, I imagine we'd test add parameter too.
2. We trust our integration tests, and any bug this mock would catch would be caught there, so it's kinda moot.